File: lib/utils/classification_display.dart
/// Maps ML-service labels to user-facing English labels for UI parity with the web app.
String typeToEnglish(Object? raw) {
  if (raw == null) return '';
  final value = raw.toString().trim();
  if (value.isEmpty) return '';

  const map = <String, String>{
    'superior': 'TOP',
    'inferior': 'BOTTOM',
    'zapatos': 'SHOES',
    'abrigo': 'COAT',
    'vestido': 'DRESS',
    'bolso': 'BAG',
    'accesorio': 'ACCESSORY',
    'joyeria': 'JEWELRY',
    'joyer\u00eda': 'JEWELRY',
    'sombrero': 'HAT',
    'cinturon': 'BELT',
    'cintur\u00f3n': 'BELT',
    'gafas': 'GLASSES',
    'desconocido': 'UNKNOWN',
  };

  final key = value.toLowerCase();
  return map[key] ?? value.replaceAll('_', ' ').toUpperCase();
}

String colorToEnglish(Object? raw) {
  if (raw == null) return '';
  final value = raw.toString().trim();
  if (value.isEmpty) return '';

  const map = <String, String>{
    'desconocido': 'Unknown',
    'negro': 'Black',
    'blanco': 'White',
    'gris': 'Gray',
    'rojo': 'Red',
    'rojo oscuro': 'Dark red',
    'azul': 'Blue',
    'azul oscuro': 'Dark blue',
    'verde': 'Green',
    'verde oscuro': 'Dark green',
    'amarillo': 'Yellow',
    'amarillo oscuro': 'Dark yellow',
    'naranja': 'Orange',
    'rosa': 'Pink',
    'beige': 'Beige',
    'marron': 'Brown',
    'marr\u00f3n': 'Brown',
    'magenta': 'Magenta',
    'multicolor': 'Multicolor',
  };

  final key = value.toLowerCase();
  return map[key] ?? value;
}

String garmentClassLabel(Object? raw) {
  if (raw == null) return 'Unknown';
  final value = raw.toString().trim();
  if (value.isEmpty) return 'Unknown';
  if (value.toLowerCase() == 'desconocido') return 'Unknown';
  return value.replaceAll('_', ' ');
}

String occasionToEnglish(Object? raw) {
  if (raw == null) return '';
  final value = raw.toString().trim();
  if (value.isEmpty) return '';

  const map = <String, String>{
    'casual': 'Casual',
    'formal': 'Formal',
    'deportivo': 'Sporty',
    'fiesta': 'Party',
    'trabajo': 'Work',
  };

  final key = value.toLowerCase();
  return map[key] ?? value.replaceAll('_', ' ');
}

String formatOccasionsEnglish(Object? raw) {
  const dash = '\u2014';
  if (raw == null) return dash;

  final items = <String>[];
  if (raw is List) {
    for (final v in raw) {
      final s = v.toString().trim();
      if (s.isNotEmpty) items.add(s);
    }
  } else {
    items.addAll(
      raw
          .toString()
          .split(',')
          .map((e) => e.trim())
          .where((e) => e.isNotEmpty),
    );
  }

  if (items.isEmpty) return dash;
  return items.map(occasionToEnglish).join(', ');
}

File: webapp/src/lib/classificationDisplay.js
/**
 * Maps ML service Spanish labels to English for UI (ViT / color detection).
 */

export function typeToEnglish(raw) {
  if (raw == null || raw === '') return ''
  const map = {
    superior: 'TOP',
    inferior: 'BOTTOM',
    zapatos: 'SHOES',
    abrigo: 'COAT',
    vestido: 'DRESS',
    bolso: 'BAG',
    accesorio: 'ACCESSORY',
    'joyería': 'JEWELRY',
    joyeria: 'JEWELRY',
    sombrero: 'HAT',
    'cinturón': 'BELT',
    cinturon: 'BELT',
    gafas: 'GLASSES',
    desconocido: 'UNKNOWN',
  }
  const s = String(raw).toLowerCase().trim()
  return map[s] || String(raw).replace(/_/g, ' ').toUpperCase()
}

export function colorToEnglish(raw) {
  if (raw == null || raw === '') return ''
  const map = {
    desconocido: 'Unknown',
    negro: 'Black',
    blanco: 'White',
    gris: 'Gray',
    rojo: 'Red',
    'rojo oscuro': 'Dark red',
    azul: 'Blue',
    'azul oscuro': 'Dark blue',
    verde: 'Green',
    'verde oscuro': 'Dark green',
    amarillo: 'Yellow',
    'amarillo oscuro': 'Dark yellow',
    naranja: 'Orange',
    rosa: 'Pink',
    beige: 'Beige',
    marrón: 'Brown',
    marron: 'Brown',
    magenta: 'Magenta',
    multicolor: 'Multicolor',
  }
  const s = String(raw).toLowerCase().trim()
  return map[s] || String(raw)
}

/** Garment class name from model (underscores → spaces); unknown in English. */
export function garmentClassLabel(raw) {
  if (raw == null || raw === '') return 'Unknown'
  const s = String(raw).toLowerCase().trim()
  if (s === 'desconocido') return 'Unknown'
  return String(raw).replace(/_/g, ' ')
}

/** Stored occasion slugs (Spanish/English keys) → English labels for wardrobe UI. */
export function occasionToEnglish(raw) {
  if (raw == null || raw === '') return ''
  const map = {
    casual: 'Casual',
    formal: 'Formal',
    deportivo: 'Sporty',
    fiesta: 'Party',
    trabajo: 'Work',
  }
  const s = String(raw).toLowerCase().trim()
  return map[s] || String(raw).replace(/_/g, ' ')
}

/** Join occasion list/array into a single English string. */
export function formatOccasionsEnglish(raw) {
  if (raw == null || raw === '') return '—'
  const arr = Array.isArray(raw)
    ? raw
    : String(raw)
        .split(',')
        .map((x) => x.trim())
        .filter(Boolean)
  if (arr.length === 0) return '—'
  return arr.map(occasionToEnglish).join(', ')
}
